ui/tests/unit/views/Profiles.spec.ts

```diff
@@ -0,0 +1,275 @@
+import { shallowMount, VueWrapper } from "@vue/test-utils";
+import Profiles from "@/views/Profiles.vue";
+import { createRouter, createWebHistory } from "vue-router";
+import * as _api from "@/api/api";
+import { Profile } from "@/types/api"
+import { processErrorMessages } from "@/helpers/errorProcessing";
+
+const api = _api as any;
+
+jest.mock("@/api/api");
+
+describe("Profiles", () => {
+    let testData: Profile[];
+    let singleTestData: Profile[];
+
+    let profileToAdd: Profile;
+    let default_profile_running: Profile;
+    let default_profile_not_running: Profile;
+
+    const mock_routes = [
+        {
+            path: "/",
+            redirect: "/item_a",
+        },
+        {
+            path: "/item_a",
+            component: {
+                template: "Welcome to item a",
+            },
+        },
+        {
+            path: "/item_b",
+            component: {
+                template: "Welcome to item b",
+            },
+        },
+        {
+            path: "/item_c",
+            component: {
+                template: "Welcome to item c",
+            },
+        },
+    ];
+    const router = createRouter({
+        history: createWebHistory(),
+        routes: mock_routes
+    })
+    let wrapper: VueWrapper<any>;
+
+    beforeEach(function() {
+        const mockRouter = {
+            push: jest.fn(),
+        };
+
+        default_profile_not_running = {
+            name: "default",
+            image: "datashield/armadillo-rserver",
+            host: "localhost",
+            port: 6311,
+            packageWhitelist: [
+                "dsBase"
+            ],
+            functionBlacklist: [],
+            datashieldSeed: "100000000",
+            options: {
+                "datashield.seed": "100000000"
+            },
+            container: {
+                tags:  [
+                    "datashield/armadillo-rserver:2.0.0",
+                    "datashield/armadillo-rserver:latest"
+                  ],
+                status: "NOT_RUNNING"
+            }
+        }
+
+        default_profile_running = JSON.parse(JSON.stringify(default_profile_not_running))
+        default_profile_running.container.status = "RUNNING"
+
+        singleTestData = [
+            {
+                name: "profile1",
```

Comment:
Armadillo errors on numeric start like name == `0` so we need to test for invalid values and fix the UI

> 13:59:23.427 [http-nio-8080-exec-3|B623EE99] TRACE o.s.w.s.m.m.a.HttpEntityMethodProcessor - Writing [{timestamp=Tue Oct 10 13:59:23 CEST 2023, status=409, error=Conflict, message=Error starting image 'datashield/armadillo-rserver:6.2.0': Status 400: {"message":"Invalid container name (0), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed"}

![image](https://github.com/molgenis/molgenis-service-armadillo/assets/371014/d6c04e2a-82a8-4b7e-af03-37378e27934c)

Reply:
The error comes from docker ... it needs minimal 2 characters ... 'x' fails similar